dts: bindings: rename nxp,kinetis-mpu compatible and add its binding

manuargue opened the pull request:
Rename "nxp,kinetis-mpu" compatible to "nxp,sysmpu" to remove the device family from its name because this hardware block is used in multiple NXP devices from different families. Also added the corresponding binding that was missing, following a similar approach of other MPU supported.

manuargue-nxp committed:
dts: bindings: rename nxp,kinetis-mpu compatible

Rename "nxp,kinetis-mpu" compatible to "nxp,sysmpu" to remove
the device family from its name.

Signed-off-by: Manuel Argüelles <manuel.arguelles@nxp.com>

manuargue-nxp committed:
dts: bindings: add nxp,sysmpu binding

Add binding for NXP SYSMPU peripheral.

Signed-off-by: Manuel Argüelles <manuel.arguelles@nxp.com>

manuargue-nxp committed:
doc: migration-guide-4.1: add note about NXP SYSMPU changes

Add note about changes for NXP SYSMPU dt binding and Kconfig option.

Signed-off-by: Manuel Argüelles <manuel.arguelles@nxp.com>

danieldegrasse commented:
While we are cleaning up the SYSMPU, can we change the Kconfig name to be `CPU_HAS_NXP_SYSMPU`? I *hope* we aren't designing other custom MPUs, but this would at least help clarify what this Kconfig does: https://github.com/zephyrproject-rtos/zephyr/blob/d04b3e03232df610833e1c5f91323a8f458b071d/arch/arm/Kconfig#L158

manuargue-nxp committed:
arch: arm: rename CPU_HAS_NXP_MPU to align with binding

Following the binding rename to "nxp,sysmpu", update the Kconfig
option to align with the binding name and to better reflect the
option's purpose.

Signed-off-by: Manuel Argüelles <manuel.arguelles@nxp.com>

manuargue commented:
> While we are cleaning up the SYSMPU, can we change the Kconfig name to be `CPU_HAS_NXP_SYSMPU`? I _hope_ we aren't designing other custom MPUs, but this would at least help clarify what this Kconfig does:

@danieldegrasse renamed as suggested to keep consistency between the binding and kconfig symbol.

dleach02 commented:
Need to fix the compliance issue

manuargue commented:
> Need to fix the compliance issue

Fixed, I didn’t notice that earlier

mmahadevan108 commented on arch/arm/Kconfig:
```diff
@@ -155,12 +155,12 @@ config CPU_HAS_ARM_MPU
 	help
-	  This option is enabled when the CPU has a Memory Protection Unit (MPU)
-	  in NXP flavor.
+	  This option is enabled when the CPU has an NXP System Memory Protection
```
Replace `an` with `a`

manuargue replied:
nitpicking: I'm not a native English speaker, so I may be wrong, but I believe `an` is correct here because `en-ex-pee` starts with a vowel sound, regardless of whether the first letter is a vowel

kartben replied:
> nitpicking: I'm not a native English speaker, so I may be wrong, but I believe `an` is correct here because `en-ex-pee` starts with a vowel sound, regardless of whether the first letter is a vowel

Yup, that's correct :)

mmahadevan108 commented on dts/bindings/mmu_mpu/nxp,sysmpu.yaml:
```diff
@@ -0,0 +1,12 @@
+# Copyright 2024 NXP
```
Should we delete the old bindings file?

manuargue replied:
there is currently no binding file
